src/noir/p2p/p2p.cpp

```diff
@@ -708,13 +712,13 @@ void p2p::plugin_initialize(const CLI::App& cli, const CLI::App& config) {
     my->heartbeat_timeout = std::chrono::milliseconds(30000 * 2);
 
     my->p2p_address = "0.0.0.0:9876"; // The actual host:port used to listen for incoming p2p connections
-    if (config.count("--p2p-listen-endpoint")) {
-      my->p2p_address = config.get_option("--p2p-listen-endpoint")->as<string>();
+    if (p2p_options->count("p2p-listen-endpoint")) {
+      my->p2p_address = p2p_options->get_option("p2p-listen-endpoint")->as<string>();
     }
 
 //    my->supplied_peers = {"127.0.0.1:9877", "127.0.0.1:9878"}; // The public endpoint of a peer node to connect to.
-    if (config.count("--p2p-peer-address")) {
-      my->supplied_peers = {config.get_option("--p2p-peer-address")->as<string>()};
+    if (p2p_options->count("p2p-peer-address")) {
+      my->supplied_peers = {p2p_options->get_option("p2p-peer-address")->as<string>()};
```

Comment:
If you bind the option to variable when adding the option, you don't need this.

```suggestion
      my->supplied_peers = p2p_options->get_option("p2p-peer-address")->as<std::vector<string>>();
```